### ISSUE TITLE
Quickest way to Remove Horizontal Padding from MUI DataGrid 

### ISSUE THREAD

**Issue by JStevens127** [mui/mui-x#4518]
### Order ID 💳

I have my API Key, is that what you need? Including a portion of that here: 155996e3860465b25d487562d4d132ccT1JERVI6MzAzMDUsRVhQSVJZPTE2NjQ4OTE5NzEwMDAsS0VZVk....

### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### The problem in depth 🔍

With MUI Table https://mui.com/material-ui/api/table/, there is a simple prop to remove horizontal padding `padding: 'none'`.

Is there an equivalent for MUI DataGrid? If not, what would be the quickest way to establish a no padding variant of my DataGrid?

Thanks!

### Your environment 🌎

<details>
  <summary>`npx @mui/envinfo`</summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

**Comment by cherniavskii** [comment]
Hey @JStevens127 
Sounds very similar to [Row heigh](https://mui.com/x/react-data-grid/rows/#row-height) and [Density selector](https://mui.com/x/react-data-grid/accessibility/#density-selector)
Is it close enough?

**Comment by kendallroth** [comment]
Yeah, taking this one step further, it would be _kinda_ nice to have `disableHeaderPadding` and `disableCellPadding` options in the column definitions, to disable it on a per-column basis. That being said, the same principles of styling can be applied through the `cellClassName` and `headerClassName`...

**Comment by cherniavskii** [comment]
> Yeah, taking this one step further, it would be kinda nice to have disableHeaderPadding and disableCellPadding options in the column definitions, to disable it on a per-column basis.

It's a trade-off - we tend to avoid adding props/options for things that can be easily customized with CSS.
Otherwise, the API surface would grow dramatically. 

That being said, for common use cases that are not that trivial, we do add props/options - like `GridColDef['headerAlign']` or `showCellRightBorder` prop.

**Comment by cherniavskii** [comment]
> to disable it on a per-column basis

This is also possible via CSS:
```tsx
<DataGrid
  {...data}
  sx={{
    "& .MuiDataGrid-cell[data-field='desk']": {
      padding: 0
    }
  }}
/>
```
